### PR TITLE
Peep79 - Allow saving of peeps after creation

### PR DESCRIPTION
**Pull request by foopod**
image

### CODE REVIEW

File: src/components/Button/ConnectButton.tsx [hunk 1]
```diff
@@ -80,7 +87,12 @@ const ConnectButton: React.FC<Props> = ({ className = 'blue' }) => {
 	}
 
 	const withoutWeb3 = async () => {
-		setProfile({ address: ethers.constants.AddressZero })
+		const sessionID = await getOffChainSession()
+		setProfile({
+			address: ethers.constants.AddressZero,
+			isOffChain: true,
```
Comment (ScreamingHawk) [inline src/components/Button/ConnectButton.tsx]: Thoughts on making address optional and using that nullability as the offChain flag? Setting address to `address(0)` was a bit of a hack.

File: src/interface/metadata.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ export const getTrait = (metadata: Trait[], category: string) => {
 
 export const getTopDescription = (metadata: Trait[]) => {
 	// FIXME This is broken with the new category system - probably need overhaul
+	return ''
```
Comment (ScreamingHawk) [inline src/interface/metadata.tsx]: Will cause lint errors due to unreachable code

File: src/interface/metadata.tsx [hunk 3]
```diff
@@ -51,6 +52,7 @@ export const getTopDescription = (metadata: Trait[]) => {
 
 export const getFullDescription = (metadata: Trait[]) => {
 	// FIXME This is broken with the new category system - probably need overhaul
+	return ''
```
Comment (ScreamingHawk) [inline src/interface/metadata.tsx]: Will cause lint errors due to unreachable code

File: src/interface/profile.tsx [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export interface Profile {
 	id?: string
 	address: string
+	isOffChain?: boolean
```
Comment (ScreamingHawk) [inline src/interface/profile.tsx]: If we keep this, can it be mandatory with a default setting of `false`?

File: src/pages/YourPeep/YourOffChainPeep.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
+import { createUseStyles } from 'react-jss'
```
Comment (ScreamingHawk) [inline src/pages/YourPeep/YourOffChainPeep.styles.tsx]: Can we import from the `YourPeep`?

```ts
import { styles as youPeepStyles } from '../YourPeep.styles`

const styles = {
  ...yourPeepStyles,
  copyID: {} //...
}

// ...
```

File: src/pages/YourPeep/YourOffChainPeep.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+		!useLocation().state
+	) {
+		return <></>
+	}
+
+	const getRandomGreeting = () => {
+		const i = Math.floor(Math.random() * greetings.length)
+		return greetings[i]
+	}
+	const { uri } = useLocation().state as propState
```
Comment (ScreamingHawk) [inline src/pages/YourPeep/YourOffChainPeep.tsx]: Can we avoid `as propState`? It's a smell that types are wrong elsewhere.

File: src/pages/YourPeep/YourOffChainPeep.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+				const peepImage = await getPeepFromURI(uri)
+				setYourPeepImage(peepImage)
+
+				// reset metadata in case they want to mint another
+				setMetadata([...DEFAULT_PEEP])
+				setProfile({ address: profile.address })
+			}
+		}
+		getYourPeep()
+	}, [])
```
Comment (ScreamingHawk) [inline src/pages/YourPeep/YourOffChainPeep.tsx]: Should `uri` be a dependency here?

File: src/pages/YourPeep/YourOffChainPeep.tsx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+					<div
+						className={`${classes.bubbleRight} ${classes.fadeInAndOut3}`}
+						onAnimationIteration={() => {
+							changePhrase(setGrannyGreeting)
+						}}
+					>
+						{grannyGreeting}
+					</div>
+				)}
+				<img className={classes.guestPeep} src={'/assets/Granny Peep.svg'} />
```
Comment (ScreamingHawk) [inline src/pages/YourPeep/YourOffChainPeep.tsx]: Can we make these a component? `BouncingPeep` and pass in the fade class, greeting, greeting setter, svg path

**Review by ScreamingHawk — CHANGES_REQUESTED** [review]

File: src/components/Button/ConnectButton.tsx (hunk 1 above)
Comment (foopod) [reply src/components/Button/ConnectButton.tsx]: sweet, done

File: src/pages/YourPeep/YourOffChainPeep.tsx (hunk 6 above)
Comment (foopod) [reply src/pages/YourPeep/YourOffChainPeep.tsx]: I don't think we can set the type of location state, for now I have made a new interface and now using it on both pages.

File: src/components/Scene/BouncingPeep.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
+import useStyles from './BouncingPeep.styles'
+import { greetings } from '../../config/text'
+
+interface Props {
+	greeting: string
```
Comment (ScreamingHawk) [inline src/components/Scene/BouncingPeep.tsx]: Can `greeting` be state managed in the component instead? Either a default value passed in or call the random function for initial state?

**Review by ScreamingHawk — APPROVED** [review]
LGTM! Nice work on this one

**Pull request merged**